### PR TITLE
ECAL AlcaReco datasets definition

### PR DESCRIPTION
**Pull request by shervin86**
Fixed runTheMatrix failures in PR #7923

### CODE REVIEW

**Comment by diguida** [comment]
please test

**Comment by shervin86** [comment]
Am I supposed to change also this file Configuration/PyReleaseValidation/python/relval_steps.py ?
Line 1074: 
```
                  '-s'            : 'ALCA:TkAlMuonIsolated+TkAlMinBias+EcalCalElectron+HcalCalIsoTrk+MuAlOverlaps',
```
EcalCalElectron should be removed or replaced by EcalCalZElectron+EcalCalWElectron+EcalCalZSCElectron?

**Comment by shervin86** [comment]
I have fixed the file Configuration/PyReleaseValidation/python/relval_steps.py
can alca double check please?

**Comment by mmusich** [comment]
@shervin86: looks ok to me. PdmV managers (@franzoni @srimanob @boudoul) who have signature over `Configuration/PyReleaseValidation` are also cc'ed, so they can comment

**Comment by mmusich** [comment]
@cmsbuild please test

**Comment by shervin86** [comment]
In the last push, there were few commits that were not supposed to be there... sorry.
```
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-03-17-0200/src/Calibration/EcalAlCaRecoProducers/plugins/EcalAlCaESAlignTrackReducer.cc:96:23: error: unused variable 'newTrackExtra' [-Werror,-Wunused-variable]
```
This error should be solved now.

**Comment by shervin86** [comment]
there seems to be something wrong in the tests: the file EcalAlCaESAlignTrackReducer.cc should not be there anymore...

**Comment by mmusich** [comment]
@shervin86 I believe these are the results before the changes.

**Comment by mmusich** [comment]
@cmsbuild please test

**Comment by shervin86** [comment]
```
----- Begin Fatal Exception 19-Mar-2015 16:56:03 CET-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing run: 1 lumi: 1 event: 1
   [1] Running path 'reconstruction_step'
   [2] Calling event method for module GoodSeedProducer/'trackerDrivenElectronSeedsTmp'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: std::vector<reco::Track>
Looking for module label: generalTracksBeforeMixing
Looking for productInstanceName: 

   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "SkipEvent = cms.untracked.vstring('ProductNotFound
```
I'm sorry but I don't really understand how it can be related...

**Comment by shervin86** [comment]
It's failing even in IB without this PR CMSSW_7_5_X_2015-03-18-0200/src , can you double check?
If it is true, can you merge this PR?

**Comment by shervin86** [comment]
Any update?

**Comment by diguida** [comment]
please test

**Comment by franzoni** [comment]
hello @shervin86 ,

the swap in of EcalCalZElectron+EcalCalWElectron is an anticipation of  https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/1643.html . Good !

There was some discussion about this during your report at the PPD General meeting: what is the update size/event of those two producers ?

Cheers, g.

**Comment by mmusich** [comment]
+1

**Comment by shervin86** [comment]
Dear Giovanni,
sorry for the late answer.

We have 2 dataformats:
  - EcalUncalElectron
  - EcalCalElectron
for each of them there are three selections:
  - W selection
  - Z selection
  - ZSC selection (one electron in tracker acceptance and the other with 2.5<|eta|<3)
for a total of 6 paths grouped in
EcalCalZElectron: Z + ZSC selection
ECalCalWElectron: W selection
EcalUncalZElectron: Z + ZSC selection
EcalUncalWElectron: W selection


Here the updated numbers:
Event size;
  EcalCalZElectron (EcalCalWElectron): 55kB/ev (40kB/ev)
  EcalUncalZElectron (EcalUncalWElectron): 95 kB/ev (80kB/ev)

Reduction factor w.r.t. PD (estimated size with 100kB/ev)
  - SingleElectron -> W selection:     24% -> 7   TB for full 2012 (using very loose ID)
  - DoubleElectron -> Z+ZSC selection: 14% -> 1.4 TB for full 2012 (using very loose ID)

I have tried to use a tigher ID for the W but it seems that running on 2012 data this will not change reduction factor significantly.
We would like to keep loose deltaEta and deltaPhi cuts to avoid inefficiencies due to mis-alignment and we have removed the E/p cut since we use the variable to calibration and we would like to avoid biases due to that.

A tightning of the selection can be considered for the 25 ns data taking to be evaluated and implemented in a further PR.
Best regards,
-- 
 							Shervin

File: Calibration/EcalAlCaRecoProducers/plugins/AlCaElectronTracksReducer.cc [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+//   ESHandle<CaloGeometry> geoHandle;
+//   iSetup.get<CaloGeometryRecord>().get(geoHandle);
+
+//   edm::ESHandle<CaloTopology> theCaloTopology;
+//   iSetup.get<CaloTopologyRecord>().get(theCaloTopology);
+//   const CaloTopology *caloTopology = theCaloTopology.product();
+  
+  // Get GSFElectrons
+  Handle<reco::GsfElectronCollection> pElectrons;
+  iEvent.getByLabel(electronLabel_, pElectrons);
```
Comment (davidlange6) [inline Calibration/EcalAlCaRecoProducers/plugins/AlCaElectronTracksReducer.cc]: please complete the consumes migration for this new plugin

File: Calibration/EcalAlCaRecoProducers/plugins/PUDumper.cc [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+
+
+//! loop over the reco particles and count leptons
+void PUDumper::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
+{
+  // get the PU collection
+  edm::Handle<std::vector<PileupSummaryInfo> > PupInfo;
+  if( !iEvent.isRealData() ){
+    iEvent.getByLabel(edm::InputTag("addPileupInfo"), PupInfo); ///\todo put it in the cfi!
```
Comment (davidlange6) [inline Calibration/EcalAlCaRecoProducers/plugins/PUDumper.cc]: and this one too..

**Comment by shervin86** [comment]
Dear David, thanks for the remarks. I've updated to one recent IB and I fixed the consumes.

**Comment by deguio** [comment]
+1
not related to DQM

**Comment by lveldere** [comment]
will you rebase?

**Comment by shervin86** [comment]
If I rebase, do I need to do another PR? And then should I wait again for tests and approvals?
How long should this take?

-- 
 							Shervin

**Comment by lveldere** [comment]
no, if you rebase and then push,
your pr will be updated automatically
Lukas

**Comment by shervin86** [comment]
Ok, since it is not possible to push the rebased branch since it is not a fast-forward, I will remove the branch and push the rebased one with the same name.
Should this work?

-- 
 							Shervin

**Comment by lveldere** [comment]
git push -f HEAD:<YOUR_BRANC_NAME>

the -f does the trick for me usually

**Comment by mmusich** [comment]
@cmsbuild please test

**Comment by mmusich** [comment]
@shervin86 now this doesn't merge anymore...

**Comment by shervin86** [comment]
It's not my fault. Somebody changed AlCaRecoStreams_cff in the meanwhile.
I can push a new rebase, it would be preferable to merge it very soon

**Comment by mmusich** [comment]
@cmsbuild please test

**Comment by mmusich** [comment]
@shervin86 we will put on hold conflicting commits changing the AlCaRecoStreams_cff

**Comment by mmusich** [comment]
+1

**Comment by shervin86** [comment]
ping orp and pdmv

**Comment by emanueledimarco** [comment]
Can we approve these changes asap? This would help in case in the meantime changes in the package will be pushed, that would need further work on our side to adjust to the changes.

**Comment by franzoni** [comment]
+1

**Comment by shervin86** [comment]
when the ORP meeting is supposed to be?

**Comment by mmusich** [comment]
@shervin86 it was supposed to happen yesterday (but skipped because of the concurring Offline/Computing workshop)
@davidlange6 can we please have it merged? Thanks

**Comment by davidlange6** [comment]
Hi Marco

Raise the issue on the ORP email in lieu of a meeting (as requested already)

Cheers-
David

**Comment by davidlange6** [comment]
+1

**Pull request merged**